sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -474,15 +479,8 @@ class SessionCatalog(
       // This function exists in the FunctionRegistry.
       true
     } else {
-      // Need to check if this function exists in the metastore.
-      try {
-        // TODO: It's better to ask external catalog if this function exists.
-        // So, we can avoid of having this hacky try/catch block.
-        getFunction(name) != null
-      } catch {
-        case _: NoSuchFunctionException => false
-        case _: AnalysisException => false // HiveExternalCatalog wraps all exceptions with it.
-      }
+      val db = name.database.getOrElse(currentDb)
+      externalCatalog.listFunctions(db,name.unquotedString).size > 0
```

Comment:
No, the right thing to do here is to implement `functionExists` in the external catalog and `HiveClientImpl`. That's what the TODO meant.